### ISSUE TITLE
[Question] How to add custom evaluation metrics?

### ISSUE THREAD

**Issue by shawpan** [tensorflow/adanet#49]
How to add custom metrics in Adanet?  tensorflow api gives the following error

```python
def custom_metrics(features, labels, predictions):
    return {
        'customMetric': 0
    }
estimator = get_adanet_model()
estimator = tf.contrib.estimator.add_metrics(estimator, custom_metrics)
```

Error: 

> UserWarning: The adanet.Estimator's model_fn should not be called directly in TRAIN mode, because its behavior is undefined outside the context of its `train` method.

**Comment by cweill** [comment]
Please upgrade your adanet to  version >= 0.5.0. You can then pass you `custom_metrics` to the `adanet.Estimator` constructor's `metrics_fn` argument: https://adanet.readthedocs.io/en/v0.5.0/adanet.html#estimator

**Comment by cweill** [comment]
Generally you want the `value` element of the tuple to be a variable, and for the `update_op` to mutate it somehow. For example look at [`tf.metrics.accuracy`](https://github.com/tensorflow/tensorflow/blob/r1.12/tensorflow/python/ops/metrics_impl.py#L418).

**Comment by shawpan** [comment]
Yes, Thanks for more specific and elaborate example `tf.metric.accuracy`.
